lib/fastlane/plugin/wpmreleasetoolkit/helper/git_helper.rb

```diff
@@ -30,31 +30,27 @@ def self.checkout_and_pull(branch)
         return false
       end
 
-      def self.cut_release_branch(branch_name)
+      # Create a new branch named `branch_name`, cutting it from branch/commit/tag `from`, and push it
+      #
+      # If the branch with that name already exists, it will instead switch to it and pull new commits.
+      #
+      # @param [String] branch_name The full name of the new branch to create, e.g "release/1.2"
+      # @param [String?] from The branch or tag from which to cut the branch from.
+      #        If `nil`, will cut the new branch from the current commit. Otherwise, will checkout that commit/branch/tag before cutting the branch.
+      # @param [Bool] push If true, will also push the branch to `origin`, tracking the upstream branch with the local one.
+      #
+      def self.create_branch(branch_name, from: nil, push: true)
```

Comment:
I like that this is now consolidated by its purpose, not it's use case – will definitely help reduce code and complexity!